packages/babel-parser/src/tokenizer/types.ts

```diff
@@ -361,7 +361,7 @@ export function tokenKeywordOrIdentifierIsKeyword(token: TokenType): boolean {
 }
 
 export function tokenIsKeywordOrIdentifier(token: TokenType): boolean {
-  return token >= tt._in && token <= tt.name;
+  return token >= tt._in && token <= tt.placeholder;
```

Comment:
Since this change will also affect `tokenIsLiteralPropertyName`, can you add a parser test for placeholder used as a property name?